### ISSUE TITLE
Support folder based configuration

### ISSUE THREAD

**Issue by unional** [DavidAnson/vscode-markdownlint#7]
```sh
root
- folderA
  + .markdownlint.json
  + some.md
- folderB
  + .markdownlist.json
  + some.md
```

This allows us to "re-purpose" the format for some specific context.
For example, when taking note:

```markdown
Here are some note about the content

> Here are some of my own observation and note on how the content applies to us.
<!-- violate MD028 here -->
> Here are some more observation notes that I want to put into another paragraph.
```
